src/main/java/io/sphere/sdk/meta/AsyncDocumentation.java

```diff
@@ -0,0 +1,1300 @@
+ </tr>
+ <tr>
+ <td>{@link java.lang.Runnable}</td>
+ <td>0</td>
+ <td>side-effects</td>
+ <td>&nbsp;</td>
+ <td>task which causes side-effects</td>
+ </tr>
+ </tbody>
+ </table>
+
+
+
+ <!--
+
+ function vs. method block, first class member, but method reference
+
+ -->
+
+
+ <h3 id="threads">Threads and the Trinity</h3>
+
+ <p>Which Thread is used for functional composition and callbacks depends on the method.
+ For {@link java.util.concurrent.CompletionStage#thenApply(Function)},
+ {@link java.util.concurrent.CompletionStage#thenAccept(Consumer)} ,
+ {@link java.util.concurrent.CompletionStage#handle(BiFunction)} etc. exist three variants:</p>
+
+ <ol>
+  <li>{@link java.util.concurrent.CompletionStage#thenApply(Function)}, no suffix, if the future is not yet completed, the the thread which calls
+ {@link java.util.concurrent.CompletableFuture#complete(Object)} is used to apply the function, if the future is completed, the thread which calls {@link java.util.concurrent.CompletionStage#thenApply(Function)} is used.
+ So this method is discouraged if you use actors or tend to block threads.</li>
+  <li>{@link java.util.concurrent.CompletionStage#thenApplyAsync(Function)} with suffix "Async" calls the function inside a Thread of {@link ForkJoinPool#commonPool()}. So you are better protected against deadlocks.</li>
+  <li>{@link java.util.concurrent.CompletionStage#thenApplyAsync(Function, Executor)}  with suffix "Async" and additional {@link Executor} parameter calls the function inside a Thread pool you specify as second parameter.</li>
+ </ol>
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationThreadTest}
+
+
+
+
+ <h3 id="error-handling">Error Handling</h3>
+
+<p>If an exception occurs with the computation, it should be propagated to the future with {@link java.util.concurrent.CompletableFuture#completeExceptionally(Throwable)} but only once in the lifetime of the future.
+ As a result the following try catch block does not make sense, since the error is inside the future which is most likely computed in another Thread:</p>
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationTest#wrongWayOfErrorHandling()}
+
+
+<h4>Recover from a failure without a new CompletionStage</h4>
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationTest#simpleRecover()}
+
+<h4>Recover from a failure with producing a new CompletionStage</h4>
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationTest#simpleRecover()}
+
+ <h4>Handle</h4>
+{@link CompletionStage#exceptionally(Function)} is like applying {@link CompletionStage#thenApply(Function)} and then {@link CompletionStage#exceptionally(Function)}:
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationTest#handleLikeExceptionallyAndThenApply()}
+
+<p>You can use the exceptions to give error specific text to the user:</p>
+
+ {@include.example io.sphere.sdk.meta.AsyncDocumentationTest#exceptionallyWithExceptionTypes()}
+
+ <p>You also need not to cover all problems:</p>
```

Comment:
I would rewrite it as: "Besides/Notice/But you do not need to cover all problems"